src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -394,17 +402,32 @@ public void stashNetworkModifications(@NonNull List<UUID> modificationUuids) {
                     .findById(modificationUuid)
                     .orElseThrow(() -> new NetworkModificationException(MODIFICATION_NOT_FOUND, String.format(MODIFICATION_NOT_FOUND_MESSAGE, modificationUuid)));
             modificationEntity.setStashed(true);
+            modificationEntity.setModificationsOrder(-1);
             this.modificationRepository.save(modificationEntity);
         }
     }
 
     @Transactional
-    public void restoreNetworkModifications(@NonNull List<UUID> modificationUuids) {
+    public void reorderNetworkModifications(UUID groupId, Boolean stashed) {
+        List<ModificationEntity> entities = this.modificationRepository.findAllStashedByGroupId(groupId, stashed);
+        int order = 0;
+        for (ModificationEntity modificationEntity : entities) {
+            modificationEntity.setModificationsOrder(order);
+            order++;
+        }
```

Comment:
you could use IntStream for cleaner reordering
    IntStream.range(0, entities.size())
             .forEach(i -> entities.get(i).setModificationsOrder(i));